src/physics/systems.rs

```diff
@@ -12,33 +12,139 @@ use rapier::geometry::{BroadPhase, ColliderBuilder, ColliderSet, NarrowPhase};
-// because systems run in parallel. This implies that if the collider creation
-// system runs before the body creation system, then it won't be able to create
-// the colliders because the related rigid-body handles don't exist yet. This
-// causes things to be initialized during multiple frames instead of just one.
 /// System responsible for creating a Rapier rigid-body and collider from their
 /// builder resources.
 pub fn create_body_and_collider_system(
     mut commands: Commands,
     mut bodies: ResMut<RigidBodySet>,
     mut colliders: ResMut<ColliderSet>,
     mut entity_maps: ResMut<EntityMaps>,
-    entity: Entity,
-    body_builder: &RigidBodyBuilder,
-    collider_builder: &ColliderBuilder,
+    standalone_body_query: Query<Without<ColliderBuilder, (Entity, &RigidBodyBuilder)>>,
+    body_and_collider_query: Query<(Entity, &RigidBodyBuilder, &ColliderBuilder)>,
+    parented_collider_query: Query<Without<RigidBodyBuilder, (Entity, &Parent, &ColliderBuilder)>>,
 ) {
-    let handle = bodies.insert(body_builder.build());
-    commands.insert_one(entity, RigidBodyHandleComponent::from(handle));
-    commands.remove_one::<RigidBodyBuilder>(entity);
-    entity_maps.bodies.insert(entity, handle);
-
-    let handle = colliders.insert(collider_builder.build(), handle, &mut bodies);
-    commands.insert_one(entity, ColliderHandleComponent::from(handle));
-    commands.remove_one::<ColliderBuilder>(entity);
-    entity_maps.colliders.insert(entity, handle);
+    for (entity, body_builder) in standalone_body_query.iter() {
+        let handle = bodies.insert(body_builder.build());
+        commands.insert_one(entity, RigidBodyHandleComponent::from(handle));
+        commands.remove_one::<RigidBodyBuilder>(entity);
+        entity_maps.bodies.insert(entity, handle);
+    }
+
+    for (entity, body_builder, collider_builder) in body_and_collider_query.iter() {
+        let handle = bodies.insert(body_builder.build());
+        commands.insert_one(entity, RigidBodyHandleComponent::from(handle));
+        commands.remove_one::<RigidBodyBuilder>(entity);
+        entity_maps.bodies.insert(entity, handle);
+
+        let handle = colliders.insert(collider_builder.build(), handle, &mut bodies);
+        commands.insert_one(entity, ColliderHandleComponent::from(handle));
+        commands.remove_one::<ColliderBuilder>(entity);
+        entity_maps.colliders.insert(entity, handle);
+    }
+
+    for (entity, parent, collider_builder) in parented_collider_query.iter() {
+        if let Some(body_handle) = entity_maps.bodies.get(&parent.0) {
+            let handle = colliders.insert(collider_builder.build(), *body_handle, &mut bodies);
+            commands.insert_one(entity, ColliderHandleComponent::from(handle));
+            commands.remove_one::<ColliderBuilder>(entity);
+            entity_maps.colliders.insert(entity, handle);
+        } // warn here? panic here? do nothing?
```

Comment:
I'm OK with doing nothing here. That way the collider will end up being created even if the rigid-body is added later to the parent entity. Though let's keep your comment anyways, in case this comes out as a bad idea in practice.